scripts/import_old_divisions.pl

```diff
@@ -0,0 +1,364 @@
+#! /usr/bin/perl -w
+#
+# Loads division data into database.
+
+use strict;
+use warnings;
+
+use FindBin;
+chdir $FindBin::Bin;
+use lib "$FindBin::Bin";
+use lib "$FindBin::Bin/../commonlib/perllib";
+
+use mySociety::Config;
+mySociety::Config::set_file("$FindBin::Bin/../conf/general");
+
+my $parldata = mySociety::Config::get('RAWDATA');
+
+use DBI; 
+use File::Slurp;
+use JSON;
+use XML::Twig;
+use File::Find;
+use Getopt::Long;
+
+# output_filter 'safe' uses entities &#nnnn; to encode characters, this is
+# the easiest/most reliable way to get the encodings correct for content
+# output with Twig's ->sprint (content, rather than attributes)
+my $outputfilter = 'safe';
+#DBI->trace(1);
+
+use vars qw($datefrom $dateto $debates
+    $lordsdebates $force $standing
+    $scotland
+);
+my $result = GetOptions(
+            "from=s" => \$datefrom,
+            "to=s" => \$dateto,
+            "debates" => \$debates,
+            "lordsdebates" => \$lordsdebates,
+            "scotland" => \$scotland,
+            "standing" => \$standing,
+            );
+
+if ((!$result) || (!$debates && !$lordsdebates && !$standing && !$scotland)) {
+    print <<END;
+
+Loads XML files division data from the parldata directory into TheyWorkForYou.
+
+--debates - process Commons Debates
+--lordsdebates - process Lords Debates
+--scotland  - process Scottish Parliament debates
+--standing - process Public Bill Commitees (Standing Committees as were)
+
+--from=YYYY-MM-DD --to=YYYY-MM-DD - process this date range
+
+END
+    exit;
+}
+
+if ($datefrom || $dateto) {
+    $datefrom = "1000-01-01" if !defined $datefrom;
+    $dateto = "9999-12-31" if !defined $dateto;
+} else {
+    $datefrom = "9999-12-31";
+    $dateto = "1000-01-01";
+}
```

Comment:
Are these not the wrong way round?

Reply:
It's the same in xml2db.pl, which there is because you could be using `--recent`, and $use could be false and then you definitely *wouldn't* want anything to match. But here, no, doesn't make sense, does it!